browsers/duckduckgo.safariextension/js/background.js

```diff
@@ -139,23 +139,32 @@ var onBeforeNavigation = function (e) {
         return
     }
 
-    // avoid redirect loops:
-    // TODO: this won't work in safari yet:
-    //if (thisTab.httpsRedirects[e.url] >= 7) {
-    //    console.log('HTTPS: cancel https upgrade. redirect limit exceeded for url: \n' + e.url)
-    //    e.preventDefault()
-    //    e.target.url = thisTab.downgradeHttpsUpgradeRequest(e)
-    //}
-
     const url = e.url
     const isMainFrame = true // always main frame in this handler
     const upgradedUrl = https.getUpgradedUrl(e.url, thisTab, isMainFrame)
 
+    // avoid redirect loops:
+    if (thisTab.httpsRedirects[url]) {
+        const downgradedUrl = thisTab.downgradeHttpsUpgradeRequest(e)
+        console.log('HTTPS: cancel https upgrade. redirect limit exceeded for: ' + url)
+
+        if (url.toLowerCase() !== downgradedUrl.toLowerCase()) {
```

Comment:
This is a new case it seems. What's it for?

Reply:
preventing infinite loops, doing `e.preventDefault()` and `e.target.url = `, even when the url is the same seems to dispatch the `beforeNavigate` even which sends it back through the handler.